### ISSUE TITLE
Teacher forcing

### ISSUE THREAD

**Issue by mmbejani** [NVIDIA/tacotron2#533]
I trained the model on small corpus (3 hr, collected by myself) and I expected that the model is going to be overfitted however it's not happened. The loss value closes to 0.2 however the generate speech in the evaluation is awful. Then, I seek for the problem and I think, I find it. When I review the training code, I found that in the training procedure in each iteration, the exact data (Mel) has been fed to model. Therefore, by probability 1, teacher forcing is used in all of the iterations. Therefore, If model just make mistake in one frame (because it didn't see this type of error) the next frames will be corrupted.
I fix this issue on my code, but after some epochs (1000) the gradient of the model becomes Nan when teacher forcing is not used (the output of the network in current state is fed for next state.

Is there any idea? Is the idea of teacher forcing is correct?

**Comment by yasntrk** [comment]
Are you cleaning your text file from numbers or etc?

**Comment by mmbejani** [comment]
yeah, I converted numbers to text rely on speech  e.g. 123 -> one hundred twenty three (as it's read in sound file).

**Comment by yasntrk** [comment]
maybe you should reduce your epoc number, it can be something like curse of dimension

**Comment by mmbejani** [comment]
I don't understand why you think that the curse of dimension is the reason. Would you elaborate?
